### ISSUE TITLE
Have option to use private repos with Jenkins-s2i

### ISSUE THREAD

**Issue by logandonley** [rht-labs/labs-ci-cd#21]
Currently, the jenkins-s2i config needs to be in a public repo, but it would be nice to have the functionality to use private repos as well (important for Labs engagements with internally hosted GitLab instances)

**Comment by InfoSec812** [comment]
This is certainly possible by modifying the inventory for the `labs-ci-cd`. In the `build_params` for the Jenkins S2I build, you would just need to modify the `SOURCE_REPOSITORY_URL` and the `SOURCE_REPOSITORY_REF`.

**Comment by logandonley** [comment]
@InfoSec812 I believe we'll need to add this as well to the template:
```json
"sourceSecret": {
                        "name": "${PIPELINE_SOURCE_SECRET}"
                    },
```
Unfortunately this would have to be a separate template, so I'm trying to think of a better way to accomplish this.

**Comment by InfoSec812** [comment]
I believe we can add it and default to an empty string without any adverse effects. I will have to test that to be sure though.

**Comment by logandonley** [comment]
@InfoSec812 Sounds good, I'll test it as well.

**Comment by sherl0cks** [comment]
closed by #25
